### PR TITLE
fix: don't throw errors in http-event-normalizer

### PR DESCRIPTION
**Pull request by robertbeal**
**What does this implement/fix? Explain your changes.**

A PR to propose changing the behaviour of the http-event-normalizer (more as a discussion point that formal PR)

Currently, the http-event-normalizer middleware [will throw an error](https://github.com/middyjs/middy/blob/main/packages/http-event-normalizer/index.js#L8) if the HTTP version cannot be established. For example if HTTP Method is missing from the event.

- (discussion point) should the middleware care? Or care enough to throw an error which breaks the code flow. My argument is that it's designed to ensure objects are instantiated/normalised,  and not to police the HTTP validity of an event object.
- It differs in behaviour (inconsistency) from the http-header-normalizer which does not throw an error if the HTTP version cannot be figured out. This PR would make the behaviour more consistent.

Impact:

It is a breaking change as the behaviour of the middleware will be different

**Does this close any currently open issues?**

No

**Any relevant logs, error output, etc?**

N/A

**Environment:**
 - Node.js: 18

**Any other comments?**

**Todo List:**
- [ ] Feature/Fix fully implemented
- [x] Amended tests
  - [x] Unit tests
  - [ ] Benchmark tests (if applicable)
- [ ] Updated relevant documentation
- [ ] Updated relevant examples

### CODE REVIEW

**Comment by robertbeal** [comment]
Just had a thought, I could amend the middleware to be configurable, as to whether it throws an error or not. That would maintain compatibility..

**Comment by willfarrell** [comment]
Thanks for the PR and good point. It should be up to the validator to throw that error. We're just about to release v5, so your timing is perfect. If you can update it be against the `release/5.0` branch I can get it merged in before it releases (any day now, waiting for the new runtime to release).

File: packages/http-event-normalizer/index.js [hunk 1]
```diff
@@ -3,13 +3,6 @@ const httpEventNormalizerMiddleware = () => {
     const { event } = request
 
     const version = pickVersion(event)
-    const isHttpEvent = isVersionHttpEvent[version]?.(event)
-    if (!isHttpEvent) {
-      throw new Error('[http-event-normalizer] Unknown http event format')
-    }
-    // VPC Lattice is an http event, however uses a different notation
-    // - query_string_parameters
-    // - is_base64_encoded
```
Comment (willfarrell) [inline packages/http-event-normalizer/index.js:12]: Can you leave this comment in please.

**Review by willfarrell — CHANGES_REQUESTED** [review]

File: packages/http-event-normalizer/index.js (hunk 1 above)
Comment (robertbeal) [reply packages/http-event-normalizer/index.js:12]: my bad, didn't intend to remove that. I'll add it back

**Comment by robertbeal** [comment]
Thanks for the prompt response. I've rebased it off `release/5.0` (hope I got the upstream/fetch/rebase right).

**Comment by robertbeal** [comment]
Will re-do with a cleaner PR git log

**Pull request closed without merging**